src/main/scala/camp/Validator.scala

```diff
@@ -0,0 +1,99 @@
+package camp
+
+/**
+  * Implement validator typeclass that should validate arbitrary value [T].
+  * @tparam T the type of the value to be validated.
+  */
+trait Validator[T] {
+  /**
+    * Validates the value.
+    * @param value value to be validated.
+    * @return Right(value) in case the value is valid, Left(message) on invalid value
+    */
+  def validate(value: T): Either[String, T]
+
+  /**
+    * And combinator.
+    * @param other validator to be combined with 'and' with this validator.
+    * @return the Right(value) only in case this validator and <code>other</code> validator returns valid value,
+    *         otherwise Left with error messages from the validator that failed.
+    */
+  def and(other: Validator[T]): Validator[T] = (value: T) => {
+    val firstValidator = this.validate(value);
+    val secondValidator = other.validate(value);
+    Either.cond(firstValidator.isRight && secondValidator.isRight,value,firstValidator.left.getOrElse("")+secondValidator.left.getOrElse(""))
+  }
+
+  /**
+    * Or combinator.
+    * @param other validator to be combined with 'or' with this validator.
+    * @return the Right(value) only in case either this validator or <code>other</code> validator returns valid value,
+    *         otherwise Left with error messages from both validators.
+    */
+  def or(other: Validator[T]): Validator[T]  = (value: T) => {
+    val firstValidator = this.validate(value);
+    val secondValidator = other.validate(value);
```

Comment:
the same: it would be nice to not invoke second validator if the first one returns valid value.